packages/core/lib/console.js

```diff
@@ -47,6 +47,10 @@ class Console extends EventEmitter {
     this.options = options;
 
     this.repl = null;
+    // we need to keep track of name conflicts that occur between contracts and
+    // repl context objects so as not to overwrite them - this is to prevent
+    // overwriting Node native objects like Buffer, number, etc.
+    this.replContextNameConflicts = {};
```

Comment:
What do you think of using a Map instead of an Object? I think it would enable less awkward conditional expressions.